opentracing-spring-cloud-starter-jaeger/src/test/java/io/opentracing/contrib/spring/cloud/starter/jaeger/AbstractSenderSpringTest.java

```diff
@@ -28,6 +30,12 @@ protected void assertSenderClass(Class senderClass) {
     assertThat(getTracer())
         .extracting("reporter")
         .flatExtracting("reporters")
+        .filteredOn(new Condition<Object>() {
+          @Override
+          public boolean matches(Object value) {
+            return value.getClass().equals(RemoteReporter.class);
+          }
+        })
         .extracting("sender")
```

Comment:
so this can assert on private fields?

Reply:
Yes, `extracting` from AssertJ doesn't care about the visibility modifier :). It happily works with `private` fields as well

Reply:
That's big gain if you ask me! It allows the production code to remain clean while also allowing for creating advanced test cases.

Reply:
thanks :+1: 

@jpkrohling maybe it could be used in jaeger java client  - e.g. test configuration -> tracer

Reply:
@pavolloffay yw!